tests/unit/resources/test_collection_smoke.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import botocore.session
+from botocore import xform_name
+from nose.tools import assert_true
+
+from boto3.session import Session
+from boto3.resources.model import ResourceModel
+
+
+# A list of names that are common names of a pagination parameter.
+# Note that this list is not comprehensive. It may have to be updated
+# in the future, but this covers a lot of the pagination parameters.
+COMMON_PAGINATION_PARAM_NAMES = [
+    # Most common names
+    'nextToken',
+    'NextToken',
+    'marker',
+    'Marker',
+    'NextMarker',
+    'nextPageToken',
+    'NextPageToken',
+
+    # S3 specific
+    'KeyMarker',
+    'NextKeyMarker',
+    'UploadIdMarker',
+    'NextUploadIdMarker',
+    'VersionIdMarker',
+    'NextVersionIdMarker',
+    'PartNumberMarker',
+    'NextPartNumberMarker',
+
+    # DynamoDB specific
+    'ExclusiveStartTableName',
+    'LastEvaluatedTableName',
+    'ExclusiveStartKey',
+    'LastEvaluatedKey',
+
+    # Kinesis specific
+    'ExclusiveStartShardId',
+    'ExclusiveStartStreamName',
+
+    # Route53 specific
+    'StartRecordName',
+    'NextRecordName',
+    'StartRecordType',
+    'NextRecordType',
+    'StartRecordIdentifier',
+    'NextRecordIdentifier',
+]
+
+
+def operation_looks_paginated(operation_model):
+    """Checks whether an operation looks like it can be paginated
+
+    :type operation_model: botocore.model.OperationModel
+    :param operation_model: The model for a particular operation
+
+    :returns: True if determines it can be paginated. False otherwise.
+    """
+    has_input_param = _shape_has_pagination_param(operation_model.input_shape)
+    has_output_param = _shape_has_pagination_param(
+        operation_model.output_shape)
+    # If there is a parameter in either the input or output that
+    # is used in pagination, mark the operation as paginateable.
+    return (has_input_param or has_output_param)
```

Comment:
What is the delta if you change this to an `and`?  How many false positives do we have now?